addons/web_editor/static/src/js/widgets/widgets.js

```diff
@@ -1520,6 +1521,35 @@ var LinkDialog = Dialog.extend({
             class: data.classes.replace(/float-\w+/, '') + ' o_btn_preview',
         }).html((data.label && data.label.length) ? data.label : data.url);
     },
+    /**
+     * @private
+     */
+    _adaptPageAnchor: function () {
+        var self = this;
+        var urlInputValue = this.$('#o_link_dialog_url_input').val();
+        var $pageAnchor = this.$('.o_link_dialog_page_anchor');
+        var isFromWebsite = _.str.startsWith(urlInputValue, '/');
+        var anchorsArray = ['', '#top', '#bottom'];
+        var $selectMenu = this.$('select[name="link_anchor"]');
+
+        $pageAnchor.toggleClass('d-none', !isFromWebsite);
+        if (isFromWebsite) {
+            this.$('.o_anchors_loading').removeClass('d-none');
+            $.get(window.location.origin + urlInputValue, function (response) {
```

Comment:
@JKE-be What is your opinion on this? I am still unsure (getting the anchors in javascript by fetching the URL). My main concern is that the feature will never be testable this way... but I don't know if creating a route for this is complicated for nothing.